MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/assetSentiment/AssetSentimentDao.java

```diff
@@ -80,7 +93,24 @@
             "(SELECT * FROM user_sentiments_table AS R " +
                 "WHERE L.asset_id = R.asset_id AND account_name = :accountName " +
                 "AND L.asset_type = R.asset_type)")
-    protected abstract List<AssetSentiment> getAssetsWithoutSentiment(AssetType assetType, String accountName);
+    protected abstract LiveData<List<AssetSentiment>> getAssetsWithoutSentiment(AssetType assetType,
+                                                                                String accountName);
+
+    /**
+     * Combines a list of AssetSentiments with a LiveData list of AssetSentiments. The result is
+     * stored in the given MediatorLiveData object. This function is used as the onChanged callback
+     * for the mediator, in order to combine two LiveData lists.
+     * @param mediator The MediatorLiveData where the merged list will be stored.
+     * @param values A list of AssetSentiments.
+     * @param other A LiveData object containing a list of AssetSentiments.
+     */
+    private static void combineLiveDataLists(MediatorLiveData<List<AssetSentiment>> mediator,
+                            List<AssetSentiment> values, LiveData<List<AssetSentiment>> other) {
+        if (other.getValue() != null) {
+            values.addAll(other.getValue());
```

Comment:
nit: Not a big issue here given it is a private method. Just note this modifies values and may have unintended consequences for the callers providing values. Maybe construct a fresh list and add values and other to it?

Reply:
https://effectivejava.googleplex.com/third-edition#page=252